internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -194,6 +195,27 @@ func (m DeploymentDataSource) Attributes() map[string]*pluginsdk.Schema {
 			Computed: true,
 		},
 
+		"nginx_app_protect": {
```

Comment:
Same comment here for `WebApplicationFirewallStatus`. If it's useful information for users to act on, we should get the values for those fields into the datasource.